packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -153,6 +167,31 @@ export class RichTextEditor extends FoundationElement {
         return true;
     }
 
+    /**
+     * This function load tip tap editor with provided markdown content by parsing into html
+     * @public
+     */
+    public setMarkdown(markdown: string): void {
+        if (this.$fastController.isConnected) {
```

Comment:
These guards seem incorrect to me, the users should be able to call setMarkdown and getMarkdown just fine before the element is connected to the DOM.

ie
```ts
const el = document.createElement('nimble-rich-text-editor');
el.setMarkdown('some markdown');
el.getMarkdown();
```

It seems like tiptap supports this concept: https://tiptap.dev/api/editor#element

> You can even initiate your editor before mounting it to an element. This is useful when your DOM is not yet available. 

I didn't think about it too much but conceptually I think we can have in the template:
```ts
<div class="container">
    <section class="editor-container" ${ref('editorContainer')}>
    </section>
</div>
```

```css
.editor-container {
    display: contents;
}
```

```ts
class RichTextEditor {
    constructor(){
        this.editor = document.createElement('div');
        // set classes, role, etc on the element
        this.tiptTapEditor = new Editor({element})
    }

    connectedCallback() {
        if (!this.editor.isConnected) {
            this.editorContainer.append(this.editor);
        }
        this.bindEditorEventThings();
    }
}
```

Reply:
Moved the editor initialization to the constructor and appended it to dom in `connectedCallback()`. Removed the guard in `setMarkdown()` and `getMarkdown()` methods, where the consumer can set and get markdown irrespective of dom.

Reply:
Re-opening again, unless I am mistaken I still do not see any test that asserts that the getMarkdown and setMarkdown methods actually work **before** the element is connected to the DOM.  Every test seems to connect first and then test something.

I'll mark the PR approved assuming we have this test case and it goes well. @mollykreis or @m-akinc can one of y'all verify this test case is added (that the element can be created and calling set and get markdown functions even though the element is not attached to the dom, and after attached we see the expected content)

Reply:
Added separate describe block `RichTextEditor Before DOM Connection` to check the `setMarkdown()` and `getMarkdown()` before the element attached to dom and asserted once after attached to dom.